### ISSUE TITLE
Download got stuck

### ISSUE THREAD

**Issue by Blast06** [FaisalUmair/udemy-downloader-gui#94]
I im downloading a course and the download got stuck a couple hours ago. The download speed is on 0kb/s and at 56%

Info:
-Im using the version 1.5.0
-I did not pause the download before it got stuck(now i tried to see if it works, but nothing happens).

Also. i was downloading another course, and i paused this one, then i resumed the download but it suddenly "started to download so fast" until 100%, but obviously wasnt downloading at all, when i checked the files, they were bad. This one, was in v1.4.0

**Comment by Blast06** [comment]
The course link is this one: https://www.udemy.com/ionic-crea-apps-para-android-e-ios/ (with this i had the problem where it got stuck at 56%, tomorrow i will try to download the whole course again and i'll let you know.)

That was on v1.5.0

the other problem i mention, was on v1.4.0 with this course: https://www.udemy.com/angular-2-fernando-herrera/

**Comment by saadafaq** [comment]
I have the same problem. 
Can I know the solution?

**Comment by shoryacr7** [comment]
Delete the latest built MTD file. The download resumes after that. Worked for me!

**Comment by Assaye** [comment]
Thanks so much, this is great @shoryacr7

**Comment by thornzz** [comment]
I have removed latest build MTD file but i didnt resume. Its stucking at always same file. Course link is : https://www.udemy.com/course/komple-javascript-gelistirici-kursu/

**Comment by syncsyncgit** [comment]
Same here, using version 1.80, delete the mtd file to no avail.
Course link: [https://www.udemy.com/course/the-complete-web-development-bootcamp/](url)

**Comment by adinugraha790** [comment]
Me at 0%, first file MTD to download in 
Course link: https://www.udemy.com/course/complete-python-bootcamp/

**Comment by Assaye** [comment]
Delete the MTD file and restart the download, it will work.

**Comment by adinugraha790** [comment]
But still, deleting MTD file doesn't work for that course. Another one that I tried worked normally.

**Comment by Assaye** [comment]
Sorry,
I downloaded all my courses and whenever I stuck I delete the last MTD file
and restart the downloader

**Comment by adinugraha790** [comment]
Restart you meant closing app and download it again? If that's the case, it doesn't work for my course.
No, I honestly thank you for answering my questions :)

**Comment by Assaye** [comment]
Yes, and sorry to hear that. Please contact the developer then - they are
usually kind to respond to requests.

Good luck

**Comment by adinugraha790** [comment]
What could be better? Contact him personally or kind of commenting in this section is enough? Because he is up there in this comment section..

**Comment by ak2766** [comment]
I'm trying to download the course `Dart and Flutter: The Complete Developer's Guide` and it got stuck at Subtitle of download item 386 (i.e. 96% of the course downloaded).  I checked but there was no file with an `.mtd` extension.  So I deleted the last item downloaded (386 - the progress bar was showing 386 of 401).  I then restarted the app and it downloaded the `.mp4` file and the , the `.vtt` file but got stuck at the `.srt` file.

I then attempted to skip file 386 by going to settings and setting it up to download from 387 to 401.  That still did not work.  So, I then attached `strace` to the `PID` and I see lines of the nature below scrolling rapidly on the screen:
```
poll([{fd=13, events=POLLIN}, {fd=14, events=POLLIN}, {fd=38, events=POLLIN}, {fd=53, events=POLLIN}], 4, 0) = 0 (Timeout)
recvmsg(53, {msg_namelen=0}, 0)         = -1 EAGAIN (Resource temporarily unavailable)
recvmsg(38, {msg_namelen=0}, 0)         = -1 EAGAIN (Resource temporarily unavailable)
recvmsg(53, {msg_namelen=0}, 0)         = -1 EAGAIN (Resource temporarily unavailable)
poll([{fd=13, events=POLLIN}, {fd=14, events=POLLIN}, {fd=38, events=POLLIN}, {fd=53, events=POLLIN}], 4, 24777) = 1 ([{fd=14, revents=POLLIN}])
recvmsg(53, {msg_namelen=0}, 0)         = -1 EAGAIN (Resource temporarily unavailable)
read(14, "!", 2)                        = 1
recvmsg(38, {msg_namelen=0}, 0)         = -1 EAGAIN (Resource temporarily unavailable)
recvmsg(53, {msg_namelen=0}, 0)         = -1 EAGAIN (Resource temporarily unavailable)
```
Let me know if there's more I can do to help clear this issue.

**Comment by kabosaif** [comment]
Hi frinds
I checked the boxes of skip attachments and skip subtitle in the sittings and it seems working

**Comment by WAS33M** [comment]
https://www.udemy.com/course/learn-oauth-2/
This course gets stuck at 96%. Item 50 of 52.
Please have a look at it.
I am facing problems in many courses will add them here one by one.

**Comment by dhanushraju** [comment]
Hi everyone who got their downloads stuck and there is no mtd file also 
I would suggest u skip attachments, subtitle, and resources because few resources udeler cannot trace and download ( example: a youtube link or expired link)

**Comment by ProNeethanR** [comment]
@dhanushraju Your solution worked for me i am very very thankful to you!
Guys its very simple just go to the settings and check box the 
Skip Attachments and Skip Subtitles!
Do this only if you are facing the problem of 0 kb/s 
Thanks once again to @dhanushraju
